### PR TITLE
Replaces all instances of bitfields I can with the BITFIELD macros

### PR DESCRIPTION
**Pull request by Merlin1230**
## About The Pull Request
Changes all bitfield adding/toggling/removing/checking with the macros

## Why It's Good For The Game
Not everyone knows bitwise, and it helps make code more readable.

## Changelog
:cl:
code: changed bitfield stuff to use macros
/:cl:

### CODE REVIEW

**Comment by Archanial** [comment]
If you are adding macros also implement them.

File: code/__DEFINES/bitflags.dm [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#define HAS_FLAG(bits, flag) bits & flag
+#define ADD_FLAG(bits, flag) bits |= flag
+#define REMOVE_FLAG(bits, flag) bits &= ~flag
```
Comment (ike709) [inline code/__DEFINES/bitflags.dm]:
```suggestion
#define HAS_FLAG(bits, flag) ((bits) & (flag))
#define ADD_FLAG(bits, flag) ((bits) |= (flag))
#define REMOVE_FLAG(bits, flag) ((bits) &= ~(flag))
```

**Comment by Merlin1230** [comment]
> If you are adding macros also implement them.

Sure thing, give me a bit when I'm not busy with other stuff.

**Comment by Merlin1230** [comment]
Huh, apparently there were already macros for this
image
I'm just gonna put these into place everywhere I can, in the meantime I'll set this PR to a draft while I work on that

**Comment by utilbird** [comment]
No god please no
> Not everyone knows bitwise, and it helps make code more readable.

Then they can learn it, there's a difference between making a something approachable and replacing it with a crutch. binary operations are a useful skill to have in programming and it's not hard at all to learn, it's not like you have to know the fundamentals of binary logic operations either, you can just learn by example if you genuinely have no interest in it.

**Comment by Merlin1230** [comment]
Its not like I'm hiding the code, you can still see what the macros do. Besides, if anything it makes it more approachable since there are now words that say what the bitwise is doing

**Comment by ike709** [comment]
I'm afraid Arch wasted your time. Converting existing bitwise to macros gives no benefit but it does increase compile time.

**Pull request closed without merging**